addons/api/addon/serializers/application.js

```diff
@@ -291,21 +291,23 @@ export default class ApplicationSerializer extends RESTSerializer {
     id,
     requestType
   ) {
+    const newPayload = copy(payload, true);
+
     primaryModelClass.attributes.forEach((attribute) => {
       const {
         name,
         options: { readOnly },
       } = attribute;
 
-      if (readOnly && !Object.hasOwn(payload, name)) {
-        payload[name] = null;
+      if (readOnly && !Object.prototype.hasOwnProperty.call(newPayload, name)) {
```

Comment:
If you're wondering why I don't just call `hasOwnProperty` on the object directly, it's to satisfy this [eslint rule](https://eslint.org/docs/rules/no-prototype-builtins)